eclipse/tern.eclipse.ide.core/src/tern/eclipse/ide/core/TernCorePlugin.java

```diff
@@ -81,7 +81,8 @@ public static File getTernCoreBaseDir() throws IOException {
 
 	public static File getTernBaseDir() throws IOException {
 		return new File(FileLocator.getBundleFile(Platform
-				.getBundle(ternjs.Activator.PLUGIN_ID)), "node_modules/tern");
+				.getBundle(ternjs.Activator.PLUGIN_ID)), "node_modules/tern")
+				.getCanonicalFile();

```

Comment:
When I stepped through this code in the debugger I could see that the file path was not correctly normalized for the platform or something – either that or it contained a ‘..’ segment. Whatever it was, it was ugly and this change fixed the ugliness.

 

From: Angelo [mailto:notifications@github.com] 
Sent: 06 November 2015 09:55
To: angelozerr/tern.java
Cc: demonfiddler
Subject: Re: [tern.java] [Enhancement] ternModuleInstalls to support folders within a plug-in JAR per Issue #332 (#361)

 

In eclipse/tern.eclipse.ide.core/src/tern/eclipse/ide/core/TernCorePlugin.java <https://github.com/angelozerr/tern.java/pull/361#discussion_r44120195> :

> @@ -81,7 +81,8 @@ public static File getTernCoreBaseDir() throws IOException {
>  
>       public static File getTernBaseDir() throws IOException {
>              return new File(FileLocator.getBundleFile(Platform
> -                            .getBundle(ternjs.Activator.PLUGIN_ID)), "node_modules/tern");
> +                            .getBundle(ternjs.Activator.PLUGIN_ID)), "node_modules/tern")
> +                            .getCanonicalFile();

Why calling getCanonicalFile(). here? What is the problem you are fixing?

For your information getTernBaseDir was removed and replaced with getTernRepositoryBaseDir (root folder of tern repository).

—
Reply to this email directly or view it on GitHub <https://github.com/angelozerr/tern.java/pull/361/files#r44120195> .Image removed by sender.